File: light-control/algorithms/single_color.py
from ._meta import BaseAlgorithm

class Algorithm(BaseAlgorithm):
    def __init__(self, num_pixels, settings) -> None:
        super().__init__(num_pixels, config, settings)
        color = settings['color']
        for i in range(len(self.pixels)):
            self.pixels[i].set(*color)

config = {
    "name": "Single Color",
    "options": {
        "color": {
            "title": "Color (HSV)",
            "type": "array",
            "minItems": 3,
            "maxItems": 4,
            "items": {
                "type": "number",
                "inclusiveMinimum": 0.0,
                "inclusiveMaximum": 1.0
            }
        }
    },
    "refresh_rate": 2
}

File: web/server/src/algorithms/singleColor.ts
import { BaseAlgorithm, AlgorithmConfig } from './_meta/index'

export const config: AlgorithmConfig = {
  name: 'Single Color',
  options: {
    color: {
      title: 'Color (HSV)',
      type: 'array',
      minItems: 3,
      maxItems: 4,
      items: { type: 'number', inclusiveMinimum: 0.0, inclusiveMaximum: 1.0 },
    },
  },
  refresh_rate: 2,
}

export class Algorithm extends BaseAlgorithm {
  constructor(numPixels: number, settings: Record<string, unknown>) {
    super(numPixels, config, settings)
    const color = settings['color'] as number[]
    for (const pixel of this.pixels) {
      pixel.set(color[0], color[1], color[2], color[3] ?? 0)
    }
  }
}
